albumentations/augmentations/transforms.py

```diff
@@ -126,19 +139,39 @@ def update_params(self, params, **kwargs):
         rows = params["rows"]
         cols = params["cols"]
 
-        if rows < self.min_height:
-            h_pad_top = int((self.min_height - rows) / 2.0)
-            h_pad_bottom = self.min_height - rows - h_pad_top
+        if self.min_height is not None:
+            if rows < self.min_height:
+                h_pad_top = int((self.min_height - rows) / 2.0)
+                h_pad_bottom = self.min_height - rows - h_pad_top
+            else:
+                h_pad_top = 0
+                h_pad_bottom = 0
         else:
-            h_pad_top = 0
-            h_pad_bottom = 0
+            if rows > self.pad_height_divisor:
+                pad_rows = rows % self.pad_height_divisor
+                pad_rows = self.pad_height_divisor - pad_rows if pad_rows > 0 else 0
+            else:
+                pad_rows = self.pad_height_divisor - rows
 
-        if cols < self.min_width:
-            w_pad_left = int((self.min_width - cols) / 2.0)
-            w_pad_right = self.min_width - cols - w_pad_left
+            h_pad_top = pad_rows // 2
+            h_pad_bottom = pad_rows - h_pad_top
+
+        if self.min_width is not None:
+            if cols < self.min_width:
+                w_pad_left = int((self.min_width - cols) / 2.0)
+                w_pad_right = self.min_width - cols - w_pad_left
+            else:
+                w_pad_left = 0
+                w_pad_right = 0
```

Comment:
Also I do not think that this if statement give to us real speedup, for this reason we can remove it for readability.

Reply:
This check is indeed needed. Without it, transform will work incorrectly when image is dimensions greater than min size. It this case it will return negative padding values.